### ISSUE TITLE
Only premium can login as username.

### ISSUE THREAD

**Issue by Hockeyfan360** [TotalFreedom/TotalFreedomMod#416]
While this sounds complicated, it isn't really that hard to do if you are able to implement the /premium i.

So basically, you cannot log in as a premium player if you aren't logged into minecraft with that player (You can't be using a cracked launcher to impose as that player). This will lower the number of OP imposters that grief under someone else's username.

However, IMO, if they are logging in as a player on the "Can't Ban by Username" list, they should be allowed to join.

**Comment by JeromSar** [comment]
@Hockeyfan360 Is this a bug or a feature request. The issue is really confusing.

**Comment by Hockeyfan360** [comment]
@JeromSar A feature request.

In simpler terms, you can't login as a premium player (you can only login as premium if you supply the correct credentials) unless they are on the "Cant ban by username" list in the config.

**Comment by JeromSar** [comment]
@Hockeyfan360 While the idea seems nice, the implementation is much more complicated than it seems, namely due to plugins' inability to access the Minecraft protocol and/or interact with it.

Additionally, I think that players should be able to use unused usernames regardless of said username being premium or not. I'd rather see an username-lock feature added.
